File: tests/test_formats.py
"""Tests for format detection."""

from src.core.formats import is_supported, detect_format, FormatGroup


def test_jpeg_extensions():
    assert is_supported("photo.jpg")
    assert is_supported("photo.JPEG")
    assert is_supported("photo.JPG")


def test_raw_extensions():
    assert is_supported("photo.cr2")
    assert is_supported("photo.NEF")
    assert is_supported("photo.arw")
    assert is_supported("photo.dng")


def test_heic_extensions():
    assert is_supported("photo.heic")
    assert is_supported("photo.HEIF")


def test_svg_extension():
    assert is_supported("icon.svg")


def test_unsupported():
    assert not is_supported("document.pdf")
    assert not is_supported("video.mp4")
    assert not is_supported("noext")


def test_detect_format_groups():
    assert detect_format("photo.jpg") == FormatGroup.PILLOW
    assert detect_format("photo.png") == FormatGroup.PILLOW
    assert detect_format("photo.webp") == FormatGroup.PILLOW
    assert detect_format("photo.bmp") == FormatGroup.PILLOW
    assert detect_format("photo.tif") == FormatGroup.PILLOW
    assert detect_format("photo.gif") == FormatGroup.PILLOW
    assert detect_format("photo.avif") == FormatGroup.HEIF
    assert detect_format("photo.heic") == FormatGroup.HEIF
    assert detect_format("photo.cr2") == FormatGroup.RAW
    assert detect_format("icon.svg") == FormatGroup.SVG
    assert detect_format("unknown.xyz") is None

File: tests/test_formats.rs
use omarchy_imageview::core::formats::{detect_format, is_supported, FormatGroup};

#[test]
fn test_jpeg_supported() {
    assert!(is_supported("photo.jpg"));
    assert!(is_supported("photo.JPEG"));
    assert!(is_supported("photo.JPG"));
    assert!(is_supported("photo.jpe"));
    assert!(is_supported("photo.jfif"));
}

#[test]
fn test_png_supported() {
    assert!(is_supported("image.png"));
    assert!(is_supported("image.PNG"));
}

#[test]
fn test_webp_supported() {
    assert!(is_supported("photo.webp"));
}

#[test]
fn test_easy_raster_formats() {
    for name in [
        "icon.ico", "shot.tga", "pix.qoi", "env.hdr", "tex.dds",
        "a.pbm", "a.pgm", "a.ppm", "a.pnm", "a.ff", "scene.exr", "photo.jxl",
    ] {
        assert!(is_supported(name), "{name}");
        assert_eq!(detect_format(name), Some(FormatGroup::Image), "{name}");
    }
}

#[test]
fn test_raw_not_supported() {
    // No demosaic pipeline — don't claim camera RAW.
    for name in ["photo.cr2", "photo.NEF", "photo.arw", "photo.dng", "photo.orf", "photo.raf"] {
        assert!(!is_supported(name), "{name}");
        assert_eq!(detect_format(name), None, "{name}");
    }
}

#[test]
fn test_heic_supported() {
    assert!(is_supported("photo.heic"));
    assert!(is_supported("photo.HEIF"));
    assert_eq!(detect_format("photo.heic"), Some(FormatGroup::Image));
}

#[test]
fn test_svg_supported() {
    assert!(is_supported("icon.svg"));
}

#[test]
fn test_unsupported() {
    assert!(!is_supported("document.pdf"));
    assert!(!is_supported("video.mp4"));
    assert!(!is_supported("noext"));
}

#[test]
fn test_detect_format_image() {
    assert_eq!(detect_format("photo.jpg"), Some(FormatGroup::Image));
    assert_eq!(detect_format("photo.png"), Some(FormatGroup::Image));
    assert_eq!(detect_format("photo.webp"), Some(FormatGroup::Image));
    assert_eq!(detect_format("photo.avif"), Some(FormatGroup::Image));
    assert_eq!(detect_format("photo.heic"), Some(FormatGroup::Image));
    assert_eq!(detect_format("photo.jxl"), Some(FormatGroup::Image));
}

#[test]
fn test_detect_format_svg() {
    assert_eq!(detect_format("icon.svg"), Some(FormatGroup::Svg));
}

#[test]
fn test_detect_format_none() {
    assert_eq!(detect_format("doc.pdf"), None);
    assert_eq!(detect_format("noext"), None);
}
